resources/qml/WelcomePages/AddPrinterByIpContent.qml

```diff
@@ -0,0 +1,244 @@
+                        if (hostnameField.text.trim() != "")
+                        {
+                            addPrinterByIpScreen.hasPushedAdd = true
+                            tempTimerRequest.running = true
+                        }
+                    }
+
+                    enabled: ! addPrinterByIpScreen.hasPushedAdd
+                    BusyIndicator
+                    {
+                        anchors.fill: parent
+                        running: { ! parent.enabled && ! addPrinterByIpScreen.hasSentRequest }
+                    }
+                }
+            }
+
+            Rectangle
+            {
+                width: parent.width
+                anchors.top: userInputFields.bottom
+                anchors.margins: 20
+
+                Label
+                {
+                    id: waitResponseLabel
+                    anchors.top: parent.top
+                    anchors.margins: 20
+                    font: UM.Theme.getFont("default")
+
+                    visible: { addPrinterByIpScreen.hasSentRequest && ! addPrinterByIpScreen.haveConnection }
+                    text: catalog.i18nc("@label", "The printer at this address has not responded yet.")
+                }
+
+                Rectangle
+                {
+                    id: printerInfoLabels
+                    anchors.top: parent.top
+                    anchors.margins: 20
+
+                    visible: addPrinterByIpScreen.haveConnection
+
+                    Label
+                    {
+                        id: printerNameLabel
+                        anchors.top: parent.top
+                        font: UM.Theme.getFont("large")
+
+                        text: "Davids-desktop"  // TODO: placeholder, alter after interface review.
+                    }
+
+                    GridLayout
+                    {
+                        anchors.top: printerNameLabel.bottom
+                        columns: 2
+                        columnSpacing: 20
+
+                        Text { font: UM.Theme.getFont("default"); text: "Type" }
+                        Text { font: UM.Theme.getFont("default"); text: "Ultimaker S5" }  // TODO: placeholder, alter after interface review.
+
+                        Text { font: UM.Theme.getFont("default"); text: "Firmware version" }
+                        Text { font: UM.Theme.getFont("default"); text: "4.3.3.20180529" }  // TODO: placeholder, alter after interface review.
+
+                        Text { font: UM.Theme.getFont("default"); text: "Address" }
+                        Text { font: UM.Theme.getFont("default"); text: "10.183.1.115" }  // TODO: placeholder, alter after interface review.
+                    }
+                }
+            }
+        }
+    }
+
+    Cura.PrimaryButton
+    {
+        id: backButton
+        anchors.left: parent.left
+        anchors.bottom: parent.bottom
+        anchors.margins: 40
+        text: catalog.i18nc("@button", "Back")
+        width: 140
```

Comment:
Wasn't there a default size for primary buttons?

Also, for every size that you hard-code, it needs to be multiplied with the pixel scale factor and if any text positioning depends on it also rounded to integer pixels.

Reply:
Fixed this in the other branch.